coverage/artifacts/local.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package artifacts
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+)
+
+//LocalArtifacts sub-type of Artifacts. Represent artifacts stored locally (
+// as oppose to artifacts stored in GCS bucket)
+type LocalArtifacts struct {
+	Artifacts
+}
+
+//NewLocalArtifacts constructs LocalArtifacts
+func NewLocalArtifacts(directory string, ProfileName string,
+	KeyProfileName string, CovStdoutName string) *LocalArtifacts {
+	return &LocalArtifacts{*New(
+		directory,
+		ProfileName,
+		KeyProfileName,
+		CovStdoutName)}
+}
+
+// ProfileReader create and returns a ProfileReader by opening the file stored in profile path
+func (arts *LocalArtifacts) ProfileReader() (io.ReadCloser, error) {
+	f, err := os.Open(arts.ProfilePath())
+	if err != nil {
+		logrus.Debugf("LocalArtifacts.ProfileReader(): os.Open(profilePath) error: %v", err)
```

Comment:
Won't this error be logged by the caller at some point? If you want more context, wrap the error:

```go
f, err := os.Open(arts.ProfilePath())
if err != nil {
    err = fmt.Errorf("failed to open the profile path: %v", err)
}
return f, err
```